config/discourse_defaults.conf

```diff
@@ -239,6 +239,9 @@ max_reqs_per_ip_mode = block
 # bypass rate limiting any IP resolved as a private IP
 max_reqs_rate_limit_on_private = false
 
+# use per user rate limits vs ip rate limits for users with this trust level or more.
+skip_per_ip_rate_limit_trust_level = 1
```

Comment:
Just to clarify, what does `1` represent here?

Reply:
It's trust level.

Reply:
Ahh icic. Can we validate that an incorrect trust level is not set here just to be sure we don't allow users to shoot themselves in the foot.

Reply:
I looked and didn't see an existing pattern for validating global setting. I'm not sure if validating this setting is worth the efforts because the site will continue to work no matter what this setting is set to.

Technically valid values are 0 to 4, and if someone makes this -1 or lower, it will have the exact same effect as if it was 0 which is a valid value. And anything higher than 4 will disable this new feature and make all users rate limited by IP and no one will get a special treatment. So it's pretty safe.

Reply:
Hmm I guess we can leave this out first since it is a global setting so a more technical user needs to change this.